src/components/bottom_section/buttons/PercentageButton.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+import { MAX_CHARACTERS, ZERO } from '../../../constants/Constants';
+
+const PercentageButton = ({
+  displayValue,
+  setWaitingValue,
+  setDisplayValue,
+  currentValue,
+  calculateLength,
+}) => {
+  const percentage = () => {
+    if (currentValue === ZERO) {
+      setDisplayValue(
+        calculateLength(String(Number(displayValue) / 100)) > MAX_CHARACTERS
+          ? String(Number(displayValue) / 100).slice(0, MAX_CHARACTERS)
```

Comment:
Pls extract .slice to helper function. You are using same `.slice(0, MAX_CHARACTERS)` in several components

Reply:
fixed